File: C++/Medium/RuntimeError/number-of-ways-where-square-of-number-is-equal-to-product-of-two-numbers_3.cpp
class Solution {
public:
    int numTriplets(vector<int>& nums1, vector<int>& nums2) {
        
        int n1 = nums1.size();
        int n2 = nums2.size();
        int res = 0;
        for (int i = 0; i < n1; i++) {
            for (int j = 0; j < n2; j++) {
                for (int k = j + 1; k < n2; k++) {
                    if (nums1[i] * nums1[i] == nums2[j] * nums2[k]) {
                        res++;
                    }
                }
            }
        }
        for (int i = 0; i < n2; i++) {
            for (int j = 0; j < n1; j++) {
                for (int k = j + 1; k < n1; k++) {
                    if (nums2[i] * nums2[i] == nums1[j] * nums1[k]) {
                        res++;
                    }
                }
            }
        }
        return res;
    }
};

File: Rust/Medium/WrongAnswer/number-of-ways-where-square-of-number-is-equal-to-product-of-two-numbers_3.rs
impl Solution {
    pub fn num_triplets(nums1: Vec<i32>, nums2: Vec<i32>) -> i32 {
        
        let mut ans = 0;
        for i in 0..nums1.len() {
            for j in i+1..nums1.len() {
                let mut k = 0;
                while k < nums2.len() {
                    if nums1[i] * nums1[j] == nums2[k] * nums2[k] {
                        ans += 1;
                    }
                    k += 1;
                }
            }
        }
        for i in 0..nums2.len() {
            for j in i+1..nums2.len() {
                let mut k = 0;
                while k < nums1.len() {
                    if nums2[i] * nums2[j] == nums1[k] * nums1[k] {
                        ans += 1;
                    }
                    k += 1;
                }
            }
        }
        ans
    }
}